core/app/models/spree/zone.rb

```diff
@@ -25,8 +25,7 @@ def self.potential_matching_zones(zone)
       if zone.country?
         # Match zones of the same kind with simialr countries
         joins(countries: :zones).
-          where('zone_members_spree_countries_join.zone_id = ? OR ' +
-                'spree_zones.default_tax = ?', zone.id, true).
+          where('zone_members_spree_countries_join.zone_id = ?', zone.id).
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.